code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm

```diff
@@ -0,0 +1,368 @@
+				snowball_machine_gun(5 * 8, 5)
+		if(3)
+			if(easy_attack)
+				ice_shotgun()
+			else
+				ice_shotgun(5, list(list(0, 30, 60, 90, 120, 150, 180, 210, 240, 270, 300, 330), list(-30, -15, 0, 15, 30)))
+
+/obj/projectile/frost_orb
+	name = "frost orb"
+	icon_state = "ice_1"
+	damage = 20
+	armour_penetration = 100
+	speed = 10
+	homing_turn_speed = 30
+	damage_type = BURN
+
+/obj/projectile/frost_orb/on_hit(atom/target, blocked = FALSE)
+	. = ..()
+	if(isturf(target) || isobj(target))
+		target.ex_act(EXPLODE_HEAVY)
+
+/obj/projectile/snowball
+	name = "machine-gun snowball"
+	icon_state = "nuclear_particle"
+	damage = 5
+	armour_penetration = 100
+	speed = 4
+	damage_type = BRUTE
+
+/obj/projectile/ice_blast
+	name = "ice blast"
+	icon_state = "ice_2"
+	damage = 15
+	armour_penetration = 100
+	speed = 4
+	damage_type = BRUTE
+
+/obj/projectile/ice_blast/on_hit(atom/target, blocked = FALSE)
+	. = ..()
+	if(isturf(target) || isobj(target))
+		target.ex_act(EXPLODE_HEAVY)
+
+/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner/ex_act(severity, target)
+	adjustBruteLoss(30 * severity - 120)
+	visible_message("<span class='danger'>[src] absorbs the explosion!</span>", "<span class='userdanger'>You absorb the explosion!</span>")
+	return
+
+/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner/Goto(target, delay, minimum_distance)
+	if(enraging)
+		return
+	return ..()
+
+/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner/MoveToTarget(list/possible_targets)
+	if(enraging)
+		return
+	return ..()
+
+/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner/Move()
+	if(enraging)
+		return
+	return ..()
+
+/// Shoots out homing frost orbs that explode into ice blast projectiles after a couple seconds
+/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner/proc/frost_orbs(added_delay = 10, shoot_times = 8)
+	for(var/i in 1 to shoot_times)
+		var/turf/startloc = get_turf(src)
+		var/turf/endloc = get_turf(target)
+		if(!endloc)
+			break
+		var/obj/projectile/frost_orb/P = new /obj/projectile/frost_orb(startloc)
+		P.preparePixelProjectile(endloc, startloc)
+		P.firer = src
+		if(target)
+			P.original = target
+		P.set_homing_target(target)
+		P.fire(rand(0, 360))
+		addtimer(CALLBACK(P, /obj/projectile/frost_orb/proc/orb_explosion, projectile_speed_multiplier), 20) // make the orbs home in after a second
+		SLEEP_CHECK_DEATH(added_delay)
+	SetRecoveryTime(40, 60)
+
+/// Called when the orb is exploding, shoots out projectiles
+/obj/projectile/frost_orb/proc/orb_explosion(projectile_speed_multiplier)
+	for(var/i in 0 to 5)
+		var/angle = i * 60
+		var/turf/startloc = get_turf(src)
+		var/turf/endloc = get_turf(original)
+		if(!startloc || !endloc)
+			break
+		var/obj/projectile/P = new /obj/projectile/ice_blast(startloc)
+		P.speed /= projectile_speed_multiplier
```

Comment:
Could you turn this into a multiplication?

Reply:
you want *= 1/projectile_speed_multiplier?

Reply:
No, just change the logic as to use a multiplication.
For example, doing `projectile_speed_multiplier = 1 - enraged * 0.5`
Instead of dividing by two, you are multiplying by 0.5